### DISCUSSION TITLE
Extending SK core skills - contributing to the project

### DISCUSSION THREAD

**Discussion by alexminza** [microsoft/semantic-kernel#2424] [General]
What are the principles and guidelines for contributing to this project by extending the core skills set with custom, more complex skills that are of generic utility for the developers and users of the library.

An example I was thinking of is [ConversationalRetrievalPlugin.cs](https://gist.github.com/alexminza/f05fb189de8d8eeb53ab62b35fbc4660) implemented by me while learning Semantic Kernel and LangChain, working on my own projects using these technologies. The plugin consists of both semantic and native skills to achieve a conversational skill to discuss about own data. It is based on the great [ConversationalRetrievalChain](https://github.com/langchain-ai/langchain/blob/master/libs/langchain/langchain/chains/conversational_retrieval/base.py) from LangChain.

There are many more skills that can be added to the Semantic Kernel and my questions are:
- Should these skills be made part of the core skills in the [semantic-kernel](https://github.com/microsoft/semantic-kernel) project?
- Should these skills be published as external libraries?
- If both - what are the criteria and the guidelines?

**Comment by alexminza** [comment]
@dluc, @evchaki, @shawncal what do you think?

**Comment by gitri-ms** [comment]
Hi @alexminza - this is a great question that we've been hearing more and more lately. We're in the process of updating our contributing guidelines to clarify this (see #2503). So the short answer is, we encourage you to keep ownership of the skills or plugins you write and keep them separate from the SK repo.

That said, there is definitely interest in having some sort of centralized list of community-created SK plugins so that they are easier to find and use. This has come up a few times in our community office hours so it's on our radar and we hopefully will have a better solution for that soon!
